// react-native-localize/mock/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.removeEventListener = exports.addEventListener = exports.usesMetricSystem = exports.usesAutoTimeZone = exports.usesAutoDateAndTime = exports.uses24HourClock = exports.getTimeZone = exports.getTemperatureUnit = exports.getCurrencies = exports.getCountry = exports.getCalendar = exports.getNumberFormatSettings = exports.findBestAvailableLanguage = exports.getLocales = void 0;
const getLocales = () => [
    // extend if needed, add the locales you want
    { countryCode: "US", languageTag: "en-US", languageCode: "en", isRTL: false },
    { countryCode: "FR", languageTag: "fr-FR", languageCode: "fr", isRTL: false },
];
exports.getLocales = getLocales;
// use a provided translation, or return undefined to test your fallback
const findBestAvailableLanguage = () => ({
    languageTag: "en-US",
    isRTL: false,
});
exports.findBestAvailableLanguage = findBestAvailableLanguage;
const getNumberFormatSettings = () => ({
    decimalSeparator: ".",
    groupingSeparator: ",",
});
exports.getNumberFormatSettings = getNumberFormatSettings;
const getCalendar = () => "gregorian"; // or "japanese", "buddhist"
exports.getCalendar = getCalendar;
const getCountry = () => "US"; // the country code you want
exports.getCountry = getCountry;
const getCurrencies = () => ["USD", "EUR"]; // can be empty array
exports.getCurrencies = getCurrencies;
const getTemperatureUnit = () => "celsius"; // or "fahrenheit"
exports.getTemperatureUnit = getTemperatureUnit;
const getTimeZone = () => "Europe/Paris"; // the timezone you want
exports.getTimeZone = getTimeZone;
const uses24HourClock = () => true;
exports.uses24HourClock = uses24HourClock;
const usesAutoDateAndTime = () => true;
exports.usesAutoDateAndTime = usesAutoDateAndTime;
const usesAutoTimeZone = () => true;
exports.usesAutoTimeZone = usesAutoTimeZone;
const usesMetricSystem = () => true;
exports.usesMetricSystem = usesMetricSystem;
exports.addEventListener = jest.fn();
exports.removeEventListener = jest.fn();
exports.default = {
    getCalendar: exports.getCalendar,
    getCountry: exports.getCountry,
    getCurrencies: exports.getCurrencies,
    getLocales: exports.getLocales,
    getNumberFormatSettings: exports.getNumberFormatSettings,
    getTemperatureUnit: exports.getTemperatureUnit,
    getTimeZone: exports.getTimeZone,
    uses24HourClock: exports.uses24HourClock,
    usesAutoDateAndTime: exports.usesAutoDateAndTime,
    usesAutoTimeZone: exports.usesAutoTimeZone,
    usesMetricSystem: exports.usesMetricSystem,
    findBestAvailableLanguage: exports.findBestAvailableLanguage,
    addEventListener: exports.addEventListener,
    removeEventListener: exports.removeEventListener,
};
